test/suites/drift/suite_test.go

```diff
@@ -148,20 +148,83 @@ var _ = Describe("Drift", Label("AWS"), func() {
 			g.Expect(env.Client.Get(env.Context, client.ObjectKeyFromObject(node), &v1.Node{})).To(Succeed())
 		}).WithTimeout(time.Minute).Should(Succeed())
 	})
-})
+	It("should deprovision nodes that have drifted due to securitygroup", func() {
+		env.ExpectSettingsOverridden(map[string]string{
+			"featureGates.driftEnabled": "true",
+		})
+
+		securitygroup := env.GetSecurityGroups(map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName})
+		Expect(len(securitygroup)).To(BeNumerically("==", 2))
+
+		createSecurityGroup := &ec2.CreateSecurityGroupInput{
+			GroupName:   awssdk.String("security-group-drift"),
+			Description: awssdk.String("Drift Test"),
+			VpcId:       awssdk.String(*securitygroup[0].VpcID),
+			TagSpecifications: []*ec2.TagSpecification{
+				{
+					ResourceType: awssdk.String("security-group"),
+					Tags: []*ec2.Tag{
+						{
+							Key:   awssdk.String("security-group-drift"),
+							Value: awssdk.String(settings.FromContext(env.Context).ClusterName),
+						},
+					},
+				},
+			},
+		}
+		newSecurityGroup, err := env.EC2API.CreateSecurityGroup(createSecurityGroup)
+		Expect(err).To(BeNil())
+		DeferCleanup(func() {
+			// Need to make sure that all instance launched with the security group are terminated
+			time.Sleep(time.Minute)
+			output, err := env.EC2API.DescribeNetworkInterfaces(&ec2.DescribeNetworkInterfacesInput{
+				Filters: []*ec2.Filter{{Name: awssdk.String("group-id"), Values: []*string{newSecurityGroup.GroupId}}},
+			})
+			Expect(err).To(BeNil())
+			_, err = env.EC2API.ModifyNetworkInterfaceAttribute(&ec2.ModifyNetworkInterfaceAttributeInput{
+				NetworkInterfaceId: output.NetworkInterfaces[0].NetworkInterfaceId,
+				Groups:             lo.Map(securitygroup, func(sg aws.SecurityGroup, _ int) *string { return sg.GroupId }),
+			})
+			Expect(err).To(BeNil())
+			_, err = env.EC2API.DeleteSecurityGroup(&ec2.DeleteSecurityGroupInput{
+				GroupId: newSecurityGroup.GroupId,
+			})
+			Expect(err).To(BeNil())
+		})
+		provider := awstest.AWSNodeTemplate(v1alpha1.AWSNodeTemplateSpec{AWS: v1alpha1.AWS{
+			SecurityGroupSelector: map[string]string{"aws-ids": fmt.Sprintf("%s,%s,%s", *securitygroup[0].GroupId, *securitygroup[1].GroupId, awssdk.StringValue(newSecurityGroup.GroupId))},
+			SubnetSelector:        map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName},
```

Comment:
What's the need to specify a custom AMI in this case? I think we just need the security group customization, right?